Cod3rsGrowth.Web/Controllers/CarrosController.cs

```diff
@@ -21,10 +24,41 @@ public IActionResult ObterTodos([FromQuery] FiltroCarro? filtro)
             return Ok(_servico.ObterTodos(filtro));
         }
 
+        [HttpGet("Disponiveis")]
+        public IActionResult ObterTodos()
+        {
+            var vendas = _servicoVenda.ObterTodos();
+            var carros = _servico.ObterTodos();
+            List<Carro> carrosDisponiveis = new List<Carro>();
+
+            vendas.ForEach(x => {
+                carros = carros
+                .Where(c => c.Id != x.IdDoCarroVendido)
+                .ToList();
+            });
+
+            return Ok(carros);
+        }
+
         [HttpGet("{Id}")]
         public IActionResult ObterPorId(int Id)
         {
-            return Ok(_servico.ObterPorId(Id));
+            var carroDesejado = _servico.ObterPorId(Id);
```

Comment:
O que acontece se o Id for null?

Reply:
Quando coloca id null fala que id null não é valido.